dynamo/plot/scVectorField.py

```diff
@@ -40,8 +42,200 @@
+            of quiver indicates the nuumber of data units per arrow length unit, e.g., m/s per plot width; a smaller
+            scale parameter makes the arrow longer.
+        vector: `str` (default: `velocity`)
+            Which vector type will be used for plotting, one of {'velocity', 'acceleration'} or either velocity field or
+            acceleration field will be plotted.
+        frontier: `bool` (default: `False`)
+            Whether to add the frontier. Scatter plots can be enhanced by using transparency (alpha) in order to show
+            area of high density and multiple scatter plots can be used to delineate a frontier. See matplotlib tips &
+            tricks cheatsheet (https://github.com/matplotlib/cheatsheets). Originally inspired by figures from scEU-seq
+            paper: https://science.sciencemag.org/content/367/6482/1151.
+        save_kwargs: `dict` (default: `{}`)
+            A dictionary that will passed to the save_fig function. By default it is an empty dictionary and the
+            save_fig function will use the {"path": None, "prefix": 'cell_wise_velocity', "dpi": None, "ext": 'pdf',
+            "transparent": True, "close": True, "verbose": True} as its parameters. Otherwise you can provide a
+            dictionary that properly modify those keys according to your needs.
+        s_kwargs_dict: `dict` (default: {})
+            The dictionary of the scatter arguments.
+        cell_wise_kwargs:
+            Additional parameters that will be passed to plt.quiver function
+    Returns
+    -------
+        Nothing but a cell wise quiver plot.
+    """
+
+    import matplotlib.pyplot as plt
+    from matplotlib import rcParams
+    from matplotlib.colors import to_hex
+
+    projection_dim_indexer = [x, y, z]
+    if X is not None and V is not None:
+        X = X[:, [x, y, z]]
+        V = V[:, [x, y, z]]
+    elif type(x) == str and type(y) == str:
+        if len(adata.var_names[adata.var.use_for_dynamics].intersection([x, y])) != 2:
+            raise ValueError(
+                "If you want to plot the vector flow of two genes, please make sure those two genes "
+                "belongs to dynamics genes or .var.use_for_dynamics is True."
+            )
+        X = adata[:, projection_dim_indexer].layers[ekey].A
+        V = adata[:, projection_dim_indexer].layers[vkey].A
+        layer = ekey
+    else:
+        if ("X_" + basis in adata.obsm.keys()) and (vector + "_" + basis in adata.obsm.keys()):
+            X = adata.obsm["X_" + basis][:, projection_dim_indexer]
+            V = adata.obsm[vector + "_" + basis][:, projection_dim_indexer]
+        else:
+            if "X_" + basis not in adata.obsm.keys():
+                layer, basis = basis.split("_")
+                reduceDimension(adata, layer=layer, reduction_method=basis)
+            if "kmc" not in adata.uns_keys():
```

Comment:
by default, we don't use kmc as our velocity projection method in cell_velocities anymore. we should check whether the default velocity graph returned from running cell_velocities existed and if not use that to perform velocity projection. Let me know if this is clear to you